src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setErrorMessage('Something went wrong'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const setParent = useCallback((arrayOfPeople: Person[]): Person[] => {
+    return arrayOfPeople.map(person => {
+      const mother = arrayOfPeople.find(p => p.name === person.motherName);
+      const father = arrayOfPeople.find(p => p.name === person.fatherName);
+
+      return {
+        ...person,
+        mother,
+        father,
+      };
+    });
+  }, []);
+
+  const preparedPeople = useMemo(() => {
+    return setParent(people);
+  }, [people, setParent]);
```

Comment:
The `useMemo` hook is used to memoize the result of a function. In this case, you are using it to store the result of `setParent(people)`. However, this is not necessary because `setParent` does not perform any expensive computations. It would be more efficient to call `setParent` directly when you need the result.